File: work/stellar-restomod/check.js
const fs = require('fs');
const buffer = fs.readFileSync('stellar_setup.glb');
const jsonLength = buffer.readUInt32LE(12);
const jsonBuffer = buffer.slice(20, 20 + jsonLength);
const jsonStr = jsonBuffer.toString('utf-8');
const data = JSON.parse(jsonStr);

const meshes = new Set();
const materials = new Set();

if (data.meshes) data.meshes.forEach(m => meshes.add(m.name || ''));
if (data.materials) data.materials.forEach(m => materials.add(m.name || ''));

console.log('MESHES:');
meshes.forEach(m => console.log(m));
console.log('MATERIALS:');
materials.forEach(m => console.log(m));

File: work/stellar-restomod/check.py
import json

with open('stellar_setup.glb', 'rb') as f:
    magic = f.read(4)
    version = int.from_bytes(f.read(4), 'little')
    length = int.from_bytes(f.read(4), 'little')
    json_chunk_length = int.from_bytes(f.read(4), 'little')
    json_chunk_type = f.read(4)
    json_data = f.read(json_chunk_length).decode('utf-8')
    data = json.loads(json_data)
    
    meshes = set()
    materials = set()
    for m in data.get('meshes', []):
        meshes.add(m.get('name', ''))
    for m in data.get('materials', []):
        materials.add(m.get('name', ''))
        
    print('MESHES:')
    for m in meshes: print(m)
    print('MATERIALS:')
    for m in materials: print(m)
